### PR TITLE
CU-2nvggn0 - Move math helpers to shared labs directory

### PR DESCRIPTION
**Pull request by Cardosaum**
## Issue
[CU-2nvggn0](https://app.clickup.com/t/2nvggn0)

## Description
This PR moves math helpers functions to a common `labs` directory, improving code reuse across Composable Labs pallets.

### CODE REVIEW

File: frame/clearing-house/src/lib.rs [hunk 1]
```diff
@@ -2150,8 +2148,7 @@ pub mod pallet {
 				};
 				let last_oracle_10bp =
 					last_oracle.try_div(&T::Decimal::saturating_from_integer(1000))?;
-				oracle = math::clip(
-					oracle,
+				oracle = oracle.clamp(
```
Comment (0xangelo) [inline frame/clearing-house/src/lib.rs:2151]: Haha, nice, that looks better too!

File: frame/lending/src/lib.rs [hunk 2]
```diff
@@ -130,7 +130,7 @@ pub mod pallet {
 	#[repr(transparent)]
 	pub struct MarketIndex(
 		#[cfg(test)] // to allow pattern matching in tests outside of this crate
-		pub MarketId,
+		pub  MarketId,
```
Comment (0xangelo) [inline frame/lending/src/lib.rs:133]: It seems there is an extra space here

File: frame/vamm/src/lib.rs [hunk 3]
```diff
@@ -984,7 +988,7 @@ pub mod pallet {
 
 			let price = Self::u256_to_balance(price_u256)?;
 
-			Ok(DecimalOf::<T>::from_inner(price))
+			Ok(price.into_decimal()?)
```
Comment (0xangelo) [inline frame/vamm/src/lib.rs:991]: Cool, I didn't predict it would work for unsigned decimals right off the bat

File: frame/vamm/src/lib.rs [hunk 4]
```diff
@@ -1231,15 +1221,10 @@ pub mod pallet {
 		) -> Result<DecimalOf<T>, DispatchError> {
 			let now = Self::now(now);
 			let weight_now: MomentOf<T> =
-				std::cmp::max(1_u64.into(), now.saturating_sub(last_twap_timestamp));
+				cmp::max(1_u64.into(), now.saturating_sub(last_twap_timestamp));
```
Comment (0xangelo) [inline frame/vamm/src/lib.rs:1224]: Now that I think about it, does `Ord` automatically come with `.max(...)` for types that implement it?

**Review by 0xangelo — COMMENTED** [review]
Lovely PR! I'll withhold my approval for now in case you want to change anything before/after Kevin's review. When you're ready, just ping me (so that Bors does not complain about stale reviews)

File: frame/lending/src/lib.rs (hunk 2 above)
Comment (Cardosaum) [reply frame/lending/src/lib.rs:133]: Oh, `cargo fmt` messed up haha

File: frame/vamm/src/lib.rs (hunk 3 above)
Comment (Cardosaum) [reply frame/vamm/src/lib.rs:991]: Yeah I thought the same, thankfully it just works :tm:

File: frame/vamm/src/lib.rs (hunk 4 above)
Comment (Cardosaum) [reply frame/vamm/src/lib.rs]: Oh, good one!

**Review by kevin-composable — APPROVED** [review]

**Review by 0xangelo — APPROVED** [review]

File: frame/clearing-house/src/lib.rs (hunk 1 above)
Comment (kevin-composable) [reply frame/clearing-house/src/lib.rs:2309]: rereading this now - I would change `oracle` here to `oracle_price`

**Review by kevin-composable — COMMENTED** [review]
I had one comment - let me know what you think

**Comment by kevin-composable** [comment]
also where is the function `clamp` declared?

**Review by kevin-composable — APPROVED** [review]

**Comment by Cardosaum** [comment]
bors try

**Review by 0xangelo — APPROVED** [review]

**Pull request merged**